### PR TITLE
METRON-1007 ambari integration for metron-management

### PR DESCRIPTION
**Pull request by ottobackwards**
This pr integrates the metron-management jar file, and the stellar management functions it contains with the ambari installation, thus removing the requirement to hand deploy the jar file to the /usr/metron/VERSION/lib directory before using the functions.

###Testing
Spin up Full Dev

* verify that the jar has been deployed as such:
```bash
[vagrant@node1 ~]$ cd /usr/metron/0.4.0/
[vagrant@node1 0.4.0]$ ls
bin  config  flux  lib  patterns  web
[vagrant@node1 0.4.0]$ ls lib
metron-api-0.4.0.jar     metron-data-management-0.4.0.jar     metron-enrichment-0.4.0-uber.jar  metron-parsers-0.4.0-uber.jar  metron-rest-0.4.0.jar
metron-common-0.4.0.jar  metron-elasticsearch-0.4.0-uber.jar  metron-management-0.4.0.jar       metron-pcap-backend-0.4.0.jar
```

* run stellar and execute a stellar management function

```bash
[vagrant@node1 0.4.0]$ bin/stellar -z node1:2181
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Stellar, Go!
Please note that functions are loading lazily in the background and will be unavailable until loaded fully.
{es.clustername=metron, es.ip=node1:9300, es.date.format=yyyy.MM.dd.HH, parser.error.topic=indexing}
[Stellar]>>> Functions loaded, you may refer to functions now...

[Stellar]>>> CONFIG_GET('PARSER','bro')
{
  "parserClassName":"org.apache.metron.parsers.bro.BasicBroParser",
  "sensorTopic":"bro",
  "parserConfig": {}
}
[Stellar]>>>
```

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [NA] Have you ensured that the full suite of tests and checks have been executed in the root incubating-metron folder via:
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```

- [NA] Have you written or updated unit tests and or integration tests to verify your changes?
- [NA] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [NA] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

File: metron-deployment/packaging/docker/rpm-docker/SPECS/metron.spec [hunk 1]
```diff
@@ -409,6 +428,7 @@ chkconfig --del metron-management-ui
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 %changelog
+* Thu Jun 29 2017 Add Metron Management jar - 0.4.0+1
```
Comment (JonZeolla) [inline metron-deployment/packaging/docker/rpm-docker/SPECS/metron.spec]: Perhaps this should be:

```
* Thu Jun 29 2017 Otto Fowler <otto@apache.org> - 0.4.0+1
- Added Metron Management jar
```
Or similar?  Not clean on when we use individual names/contact info or `Apache Metron <dev@metron.apache.org>`.

**Comment by ottobackwards** [comment]
@JonZeolla the name should be all set.  Any other feedback?

**Comment by dlyle65535** [comment]
+1 (inspection). Nice one, Otto, thanks. Just bumped into this yesterday.

File: metron-deployment/packaging/docker/rpm-docker/SPECS/metron.spec [hunk 2]
```diff
@@ -274,6 +276,23 @@ This package installs the Metron Indexing files
 %{metron_home}/config/zeppelin/metron/metron-ip-report.json
 %{metron_home}/config/zeppelin/metron/metron-connection-volume-report.json
 
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (JonZeolla) [inline metron-deployment/packaging/docker/rpm-docker/SPECS/metron.spec]: Even though it doesn't functionally matter maybe we should trim off those squares?

File: metron-platform/metron-management/README.md [hunk 3]
```diff
@@ -244,10 +244,12 @@ The functions are split roughly into a few sections:
   * Returns: The String representation of the enrichment config
 
 ## Deployment Instructions
-
-Deployment is as simple as dropping the jar created by this project into
-`$METRON_HOME/lib` and starting the Stellar shell via
-`$METRON_HOME/bin/stellar`
+* Ambari
```
Comment (JonZeolla) [inline metron-platform/metron-management/README.md]: More detail would be helpful here.  Perhaps "Clusters installed via the Ambari management pack (default)" and "Manual installation"?

File: metron-deployment/packaging/docker/rpm-docker/SPECS/metron.spec (hunk 2 above)
Comment (ottobackwards) [inline metron-deployment/packaging/docker/rpm-docker/SPECS/metron.spec]: I am sorry I don't know what you mean

File: metron-platform/metron-management/README.md (hunk 3 above)
Comment (ottobackwards) [inline metron-platform/metron-management/README.md]: ok

File: metron-deployment/packaging/docker/rpm-docker/SPECS/metron.spec (hunk 2 above)
Comment (JonZeolla) [inline metron-deployment/packaging/docker/rpm-docker/SPECS/metron.spec]: The "  " at the end of this line.

**Comment by JonZeolla** [comment]
+1 via inspection, pending my final outstanding comment.

File: metron-deployment/packaging/docker/rpm-docker/SPECS/metron.spec (hunk 2 above)
Comment (ottobackwards) [inline metron-deployment/packaging/docker/rpm-docker/SPECS/metron.spec]: I don't have that in my editor, do you have it in vim or something?  or is it just from what you see on the web?

line 279?

Comment (JonZeolla) [inline metron-deployment/packaging/docker/rpm-docker/SPECS/metron.spec]: See [here](https://github.com/ottobackwards/metron/pull/4).

Comment (ottobackwards) [inline metron-deployment/packaging/docker/rpm-docker/SPECS/metron.spec]: I still can't see what you mean, I will take your word for it

**Comment by JonZeolla** [comment]
Ok final +1 (via inspection), thanks!

**Pull request closed without merging**